File: 프로그래머스/lv1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.py
def solution(price, money, count):
    need = ((price + (price * count)) * count) / 2

    if money - need > 0:
        return 0

    return need - money
    
    


# class Solution {
#     public long solution(int price, int money, int count) {        
#         long need = ((price + (price * count)) * count) / 2;
        
#         if (money - need > 0) {
#             return 0;
#         }
        
#         return need - money;
#     }
# }

File: 프로그래머스/lv1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {        
        long first = price;
        long last = (long)price * count;
        
        
        long need = ((first + last) * count) / 2;
        
        
        return Math.max(need - (long)money, 0);
    }
}